cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -35,6 +34,93 @@ namespace lists {
 namespace detail {
 namespace {
 using offset_type = lists_column_view::offset_type;
+
+/**
+ * @brief Performs an equality comparison between two entries in a lists column
+ *
+ * For the two elements that are in the same list in the lists column, they will always be
+ * considered as different. If they are from the same list and their type is one of floating
+ * point types, this functor will return the same comparison result as
+ * `cudf::element_equality_comparator`. For floating point types, entries holding NaN value will
+ * be considered as different.
+ *
+ * @tparam Type The data type of entries
+ */
+template <class Type>
+class list_entry_comparator {
+ public:
+  __host__ __device__ list_entry_comparator(offset_type const* list_offsets,
```

Comment:
You might want to double check if `__host__ __device__` are strictly required for constructors. My guess is that this can be skipped.